pyqtgraph/functions.py

```diff
@@ -253,45 +253,26 @@ def mkColor(*args):
             c = args[0]
             if len(c) == 1:
                 try:
-                    return Colors[c]
+                    return QtGui.QColor(Colors[c])  # return copy
                 except KeyError:
                     raise ValueError('No color named "%s"' % c)
-            have_alpha = len(c) in [5, 9] and c[0] == '#'  # "#RGBA" and "#RRGGBBAA"
-            if not have_alpha:
-                # try parsing SVG named colors, including "#RGB" and "#RRGGBB".
-                # note that QColor.setNamedColor() treats a 9-char hex string as "#AARRGGBB".
-                qcol = QtGui.QColor()
-                qcol.setNamedColor(c)
+            # match hex color codes
+            match = re.match(r"#([0-9a-fA-F]{3,8})", c)
```

Comment:
My personal impression is that it might be better to try avoiding the overhead of running the regex parser just to confirm the hex nature of the input string. As a relatively low level function, I am sure that there's a use case where `mkColor` gets called a lot.
Maybe we can get away with just checking the lead '#' character and *requiring* that this is followed by a valid hex sequence. Ideally, we'll fail with a readable error message if the conversion fails later?